### PR TITLE
chore: wrap relationship fields in ArrayField

### PR DESCRIPTION
**Pull request by hein-j**
*Description of changes:*
- Wrap relationship fields in `ArrayField`
- Init snapshot testing for `HAS_ONE` relationship

Delimitations:
- Since `Autocomplete` doesn't exist yet, this will break E2E.
- Tasks to map props to the `Autocomplete` and `ArrayField` outstanding.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by cwoolum — APPROVED** [review]
LGTM

File: packages/codegen-ui-react/lib/forms/form-renderer-helper.ts [hunk 1]
```diff
@@ -55,6 +55,8 @@ import {
   setStateExpression,
 } from './form-state';
 
+export const shouldWrapInArrayField = (config: FieldConfigMetadata) => config.isArray || config.relationship;
```
Comment (SwaySway) [inline packages/codegen-ui-react/lib/forms/form-renderer-helper.ts:58]: nit: I wonder if we should start splitting up functionality into separate functions for better readability (ex. relationship helpers into their own files)

**Review by SwaySway — APPROVED** [review]
🚀

File: packages/codegen-ui-react/lib/forms/form-renderer-helper.ts (hunk 1 above)
Comment (hein-j) [reply packages/codegen-ui-react/lib/forms/form-renderer-helper.ts:58]: I agree this form-renderer-helper file is getting unwieldy. Will take note to split this up in a separate PR.

**Pull request merged**